core/src/main/scala/kafka/server/ReplicaManager.scala

```diff
@@ -942,7 +942,7 @@ class ReplicaManager(val config: KafkaConfig,
                                requiredAcks: Short): Map[TopicPartition, LogAppendResult] = {
     val traceEnabled = isTraceEnabled
     def processFailedRecord(topicPartition: TopicPartition, t: Throwable) = {
-      val logStartOffset = onlinePartition(topicPartition).map(_.logStartOffset).getOrElse(-1)
+      val logStartOffset: Long = onlinePartition(topicPartition).map(_.logStartOffset).getOrElse(-1)
```

Comment:
Might be simpler to use -1L?

Reply:
Doh! I was confused about why it had failed.  Will fix.